test/drivers/second_quantization/test_driver_molecule.py

```diff
@@ -55,6 +57,7 @@ def test_construct(self):
                 degrees_of_freedom=[stretch],
                 masses=[0.7, 0.8],
             )
+            self.assertEqual(mol.units, UnitsType.ANGSTROM)
```

Comment:
It seems that while Bohr was added there is nothing that tests this, say explicitly creating a molecule in Bohr units that this. Maybe one of these could be altered to Bohr and perhaps we could pass Angstom in explicitly too in one of these so it does not rely on default?.

In looking at the logic here we have the property `units` as per mol.units yet the constructor arg is `unit` - maybe since the latter has just been added `units` would be better for the constructor param to keep things consistent.